backend/tests/services/execution/test_recovery_service.py

```
@@ -196,6 +196,141 @@ async def test_recover_with_archive_continues_when_restore_fails():
+            recovery_module.archive_service,
+            "restore_workspace",
+            AsyncMock(
+                return_value={
+                    "success": True,
+                    "session_restored": False,
+                    "git_restored": True,
+                }
+            ),
+        ),
+    ):
+        with pytest.raises(RuntimeError, match="session context.*not restored"):
```

Ruff (0.16.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)